scripts/events_module/relationship/romantic_events.py

```diff
@@ -681,7 +681,7 @@ def prepare_relationship_string(mate_string, cat_from, cat_to):
             mate_string = mate_string.replace("[m_c_mates]", mate_name_string)
 
         if "[r_c_mates]" in mate_string:
-            mate_names = [str(cat_to.fetch_cat(mate_id).name) for mate_id in cat_to.mate]
+            mate_names = [str(cat_to.fetch_cat(mate_id).name) for mate_id in cat_to.mate if not cat_to.fetch_cat(mate_id).dead and not cat_to.fetch_cat(mate_id).outside]
```

Comment:
In general ok, but the thing is, that .fetch_cat might return an None if the id is not found, which would then lead to an error because None doesn't have variables like .name or .dead or .outside.
Not sure if you can change it in this PR to prevent such errors.